File: petclinic-data/src/main/java/xyz/haff/petclinic/models/PersonalData.java
package xyz.haff.petclinic.models;

import lombok.*;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;

// TODO: Maybe add address here too

@Entity
@Getter
@Setter
@ToString(onlyExplicitlyIncluded = true)
@NoArgsConstructor
@SuperBuilder
public class PersonalData extends AbstractBaseEntity {
    private String firstName;
    private String lastName;

    @OneToOne(cascade = { CascadeType.MERGE, CascadeType.PERSIST })
    @ToString.Include
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    @Builder.Default private User user = User.builder().build();

    @ToString.Include
    public String fullName() {
        return String.format("%s %s", firstName, lastName);
    }
}

File: petclinic-data/src/main/kotlin/xyz/haff/petclinic/models/PersonalData.kt
package xyz.haff.petclinic.models

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Version
import java.util.*

// TODO: Or rather, just `Person`?

data class PersonalData @JvmOverloads constructor(
    val baseEntity: BaseEntity = BaseEntity(),
    val firstName: String,
    val lastName: String,
    val user: User
): Entity by baseEntity {
    fun fullName(): String = "$firstName $lastName"
}